chain/network-primitives/src/types.rs

```diff
@@ -1022,6 +1024,345 @@ impl PartialEncodedChunkForwardMsg {
     }
 }
 
+/// Represents edge between two nodes. Unlike `Edge` it doesn't contain signatures.
+#[derive(Hash, Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct SimpleEdge {
```

Comment:
@bowenwang1996 @pmnoxx Can you sync up on what belongs to the public interface here? There was a previous conversation about it: https://github.com/near/nearcore/pull/4651#discussion_r687945878

Reply:
This should not belong to the public interface. `SimpleEdge` should be purely internal to the network implementation itself

Reply:
@frol @bowenwang1996 I decided to move those structures back to `network`. I'll make `jsonrpc` depending on `network` as an optional dependency under `adversarial` flag.

Reply:
Thanks, done.